### PR TITLE
feat: add alteration notice API (HL-1158)

### PR DESCRIPTION
**Pull request by EmiliaMakelaVincit**
## Description :sparkles:
Implements the creation and modification interfaces for application alterations.

### CODE REVIEW

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2856.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2856.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2856.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2856.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2856.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2856.dev.hel.ninja :laughing::tada::tada::tada:

File: backend/benefit/applications/api/v1/serializers/application_alteration.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+        application = merged_data["application"]
+
+        if settings.NEXT_PUBLIC_MOCK_FLAG:
+            if not (user and user.is_authenticated):
+                user = get_user_model().objects.all().order_by("username").first()
+
+        if not user.is_handler():
+            company = get_company_from_request(request)
+            print(company, application.company)
```
Comment (rikuke) [inline backend/benefit/applications/api/v1/serializers/application_alteration.py:133]: These print statements can probably be removed ?

File: backend/benefit/applications/api/v1/serializers/application_alteration.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+            print(company.id, application.company.id)
```
Comment (rikuke) [inline backend/benefit/applications/api/v1/serializers/application_alteration.py:134]: Print statement

**Review by rikuke — CHANGES_REQUESTED** [review]

**Review by rikuke — APPROVED** [review]

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2856.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2856.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2856.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2856.dev.hel.ninja :laughing::tada::tada::tada:

**Pull request merged**